base/compiler/typelattice.jl

```diff
@@ -45,6 +44,15 @@ struct Conditional
     end
 end
 
+# # similar to `Conditional`, but conveys inter-procedural constrains imposed on call arguments
+# struct InterConditional
+#     slot::Int
+#     vtype
+#     elsetype
+# end
+import Core: InterConditional
+const ConditionalWrapper = Union{Conditional,InterConditional}
```

Comment:
I'm not sure I like `ConditionalWrapper`. Maybe `AnyConditional`?

Reply:
renamed